server/src/internalClusterTest/java/org/elasticsearch/discovery/StableMasterDisruptionIT.java

```diff
@@ -610,7 +597,14 @@ public void testNoQuorum() throws Exception {
         networkDisconnect.startDisrupting();
         internalCluster().stopNode(firstMasterNode);
         for (String nonActiveMasterNode : nonActiveMasterNodes) {
-            assertMasterStability(internalCluster().client(nonActiveMasterNode), HealthStatus.RED, "unable to form a quorum");
+            assertMasterStability(
+                internalCluster().client(nonActiveMasterNode),
+                HealthStatus.RED,
+                anyOf(
+                    containsString("unable to form a quorum"),
+                    containsString("No master node observed in the last 1s") // happens if master node has not replied within 1s
```

Comment:
I was able to force this situation by adding a sleep before https://github.com/elastic/elasticsearch/blob/352a688b041746a669879022b6b1934f8a011892/server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java#L664 that populates response map that is later used to build an error message. I suspect the failure is happening when test is running on a slow hardware and a `nonActiveMasterNode` is not receiving a response in time.